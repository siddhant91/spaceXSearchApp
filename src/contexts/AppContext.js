import { useState, createContext } from 'react';
import { node } from 'prop-types';

const AppContext = createContext();

const AppProvider = ({ children }) => {
	const [loaderVisible, setLoaderVisible] = useState(false);

	return (
		<AppContext.Provider value={[loaderVisible, setLoaderVisible]}>{children}</AppContext.Provider>
	);
};

AppProvider.propTypes = {
	children: node.isRequired,
};

export { AppContext, AppProvider };
