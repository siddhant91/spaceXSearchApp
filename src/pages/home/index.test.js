import { render, cleanup, act, screen, waitFor } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import Home from './index';
import { AppProvider } from '../../contexts/AppContext';
import mockShuttlesList from '../../../mockData/shuttlesList';

afterEach(cleanup);
const mock = new MockAdapter(axios);

const renderComponent = (data = {}, mockError = false) => {
	act(() => {
		const mockData = data;
		if (mockError) {
			const mockErrorObj = {
				status: 400,
				errorMessage: 'bad Request',
			};
			mock.onGet().reply(400, mockErrorObj);
		} else {
			mock.onGet().reply(200, mockData);
		}
	});

	return (
		<AppProvider>
			<Home />
		</AppProvider>
	);
};

it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent());
	expect(asFragment()).toMatchSnapshot();
});

it('renders the Filter title ', async () => {
	const { findByText } = render(renderComponent());
	expect(await findByText('Filters')).toBeInTheDocument();
});

it('fetches the empty data ', async () => {
	render(renderComponent());
	expect(await screen.findByText('No Cards available')).toBeInTheDocument();
});

it('fetches the data ', async () => {
	render(renderComponent(mockShuttlesList));
	const shuttleCards = await screen.findAllByTestId('shuttle-card');
	await waitFor(() => {
		expect(shuttleCards).toHaveLength(2);
	});
});
