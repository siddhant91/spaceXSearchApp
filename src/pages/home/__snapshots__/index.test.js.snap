// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spacex-homepage d-flex w-100 flex-column flex-sm-row"
  >
    <div
      class="spacex-loader spacex-loader--wrapper"
    >
      <div
        class="spacex-loader--container"
      />
    </div>
    <div
      class="spacex-homepage__filters"
    >
      <h2>
        Filters
      </h2>
      <div
        class="spacex-filter-list"
      >
        <div
          class="spacex-filter-list--title"
        >
          <h3>
            Launch Year
          </h3>
        </div>
        <div
          class="d-flex flex-wrap justify-content-between spacex-filter-list--items"
          data-testid="filter-list-container"
        >
          <button
            aria-label="2006"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2006"
            type="button"
          >
            2006
          </button>
          <button
            aria-label="2007"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2007"
            type="button"
          >
            2007
          </button>
          <button
            aria-label="2008"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2008"
            type="button"
          >
            2008
          </button>
          <button
            aria-label="2009"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2009"
            type="button"
          >
            2009
          </button>
          <button
            aria-label="2010"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2010"
            type="button"
          >
            2010
          </button>
          <button
            aria-label="2011"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2011"
            type="button"
          >
            2011
          </button>
          <button
            aria-label="2012"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2012"
            type="button"
          >
            2012
          </button>
          <button
            aria-label="2013"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2013"
            type="button"
          >
            2013
          </button>
          <button
            aria-label="2014"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2014"
            type="button"
          >
            2014
          </button>
          <button
            aria-label="2015"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2015"
            type="button"
          >
            2015
          </button>
          <button
            aria-label="2016"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2016"
            type="button"
          >
            2016
          </button>
          <button
            aria-label="2017"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2017"
            type="button"
          >
            2017
          </button>
          <button
            aria-label="2018"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2018"
            type="button"
          >
            2018
          </button>
          <button
            aria-label="2019"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2019"
            type="button"
          >
            2019
          </button>
          <button
            aria-label="2020"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="2020"
            type="button"
          >
            2020
          </button>
        </div>
      </div>
      <div
        class="spacex-filter-list"
      >
        <div
          class="spacex-filter-list--title"
        >
          <h3>
            Successful Launch
          </h3>
        </div>
        <div
          class="d-flex flex-wrap justify-content-between spacex-filter-list--items"
          data-testid="filter-list-container"
        >
          <button
            aria-label="True"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="true"
            type="button"
          >
            True
          </button>
          <button
            aria-label="False"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="false"
            type="button"
          >
            False
          </button>
        </div>
      </div>
      <div
        class="spacex-filter-list"
      >
        <div
          class="spacex-filter-list--title"
        >
          <h3>
            Successful Landing
          </h3>
        </div>
        <div
          class="d-flex flex-wrap justify-content-between spacex-filter-list--items"
          data-testid="filter-list-container"
        >
          <button
            aria-label="True"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="true"
            type="button"
          >
            True
          </button>
          <button
            aria-label="False"
            class="spacex-button-component"
            data-testid="filter-button"
            data-value="false"
            type="button"
          >
            False
          </button>
        </div>
      </div>
    </div>
    <div
      class="spacex-homepage--cards d-flex flex-wrap"
    >
      <p>
        No Cards available
      </p>
    </div>
  </div>
</DocumentFragment>
`;
